### PR TITLE
Expose connect method to open a socket to ABQ.

### PR DESCRIPTION
**Pull request by doxavore**

### CODE REVIEW

File: src/socket.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Socket } from 'net'
+import { AbqConfiguration } from './configuration'
+
+export async function connect(abqConfig: AbqConfiguration): Promise<Socket> {
```
Comment (michaelglass) [inline src/socket.ts:4]: nice!

**Review by michaelglass — APPROVED** [review]

File: src/socket.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  if (!abqConfig.enabled) {
+    throw new Error('abq must be enabled to connect')
+  }
+
+  const socket = new Socket()
+
+  return await new Promise(resolve => {
```
Comment (michaelglass) [inline src/socket.ts:11]: 💭 wonder if this should also handle the spawnedMessage call

Comment (doxavore) [reply src/socket.ts:11]: This package is _very_ low-level right now. I think I agree though... we should review once we have two consumers of it to hone in on what the right level of abstraction is.

**Pull request merged**